dotCMS/src/main/java/com/dotmarketing/db/DataSourceStrategyProvider.java

```diff
@@ -59,38 +59,47 @@ public DataSource get()
 
         final String providerClassName = getCustomDataSourceProvider();
 
-        if (!UtilMethods.isSet(providerClassName)) {
-            if (getDBPropertiesInstance()
-                    .existsDBPropertiesFile()) {
-                defaultDataSource = getDBPropertiesInstance()
-                        .apply();
-                Logger.info(DbConnectionFactory.class,
-                        "Datasource loaded from db.properties file");
-            } else if (systemEnvironmentProperties.getVariable("connection_db_base_url") != null) {
-                defaultDataSource = getSystemEnvDataSourceInstance()
-                        .apply();
-                Logger.info(DbConnectionFactory.class,
-                        "Datasource loaded from system environment");
-            } else if (getDockerSecretDataSourceInstance().dockerSecretPathExists()){
-                defaultDataSource = getDockerSecretDataSourceInstance()
-                        .apply();
+        try {
+
+            if (!UtilMethods.isSet(providerClassName)) {
+                if (getDBPropertiesInstance()
+                        .existsDBPropertiesFile()) {
+                    defaultDataSource = getDBPropertiesInstance()
+                            .apply();
+                    Logger.info(DbConnectionFactory.class,
+                            "Datasource loaded from db.properties file");
+                } else if (systemEnvironmentProperties.getVariable("connection_db_base_url")
+                        != null) {
+                    defaultDataSource = getSystemEnvDataSourceInstance()
+                            .apply();
+                    Logger.info(DbConnectionFactory.class,
+                            "Datasource loaded from system environment");
+                } else if (getDockerSecretDataSourceInstance().dockerSecretPathExists()) {
+                    defaultDataSource = getDockerSecretDataSourceInstance()
+                            .apply();
+                    Logger.info(DbConnectionFactory.class,
+                            "Datasource loaded from Docker Secret");
+                }
+            } else {
+                DotDataSourceStrategy customStrategy = ((Class<DotDataSourceStrategy>) Class
+                        .forName(providerClassName)).newInstance();
+                defaultDataSource = customStrategy.apply();
+
                 Logger.info(DbConnectionFactory.class,
-                        "Datasource loaded from Docker Secret");
+                        "Datasource loaded using custom class " + providerClassName);
             }
 
+        } catch(Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5092420)